### PR TITLE
{bio}[GCC/12.2.0]  SPAdes v3.15.4, STAR v2.7.10b w/ Python 3.10.8

### PR DESCRIPTION
**Pull request by eylenth**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
himem04.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6230 CPU @ 2.10GHz (cascadelake), Python 3.6.8
See https://gist.github.com/jfgrimm/0bf1b5ac6da127c5f77711bf91c5071f for a full test report.

File: easybuild/easyconfigs/s/STAR/STAR-2.7.10b-GCC-12.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+homepage = 'https://github.com/alexdobin/STAR'
+description = "STAR aligns RNA-seq reads to a reference genome using uncompressed suffix arrays."
+
+toolchain = {'name': 'GCC', 'version': '12.2.0'}
+toolchainopts = {'openmp': True}
+
+source_urls = ['https://github.com/alexdobin/STAR/archive/']
+sources = ['%(version)s.tar.gz']
+checksums = ['0d1b71de6c5be1c5d90b32130d2abcd5785a4fc7c1e9bf19cc391947f2dc46e5']
```
Comment (jfgrimm) [inline easybuild/easyconfigs/s/STAR/STAR-2.7.10b-GCC-12.2.0.eb:20]: github_account = 'alexdobin'
source_urls = [GITHUB_SOURCE]
sources = ['%(version)s.tar.gz']
patches = ['STAR-%(version)s_use-external-htslib.patch']
checksums = [
    {'2.7.10b.tar.gz': '0d1b71de6c5be1c5d90b32130d2abcd5785a4fc7c1e9bf19cc391947f2dc46e5'},
    {'STAR-2.7.10b_use-external-htslib.patch': '1d8a24532a384b3065001c672e486b74094f162cb98829d2ec961401215de7da'},
]

File: easybuild/easyconfigs/s/STAR/STAR-2.7.10b-GCC-12.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+dependencies = [
+    ('zlib', '1.2.12'),
+]
```
Comment (jfgrimm) [inline easybuild/easyconfigs/s/STAR/STAR-2.7.10b-GCC-12.2.0.eb:24]:
```suggestion
dependencies = [
    ('HTSlib', '1.17'),
    ('zlib', '1.2.12'),
]
```

File: easybuild/easyconfigs/s/STAR/STAR-2.7.10b-GCC-12.2.0.eb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+start_dir = 'source'
+
+buildopts = ' STAR && make STARlong'
+
+parallel = 1
```
Comment (jfgrimm) [inline easybuild/easyconfigs/s/STAR/STAR-2.7.10b-GCC-12.2.0.eb:30]:
```suggestion
```

File: easybuild/easyconfigs/s/STAR/STAR-2.7.10b-GCC-12.2.0.eb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+files_to_copy = [
+    (['source/STAR', 'source/STARlong'], 'bin'),
```
Comment (jfgrimm) [inline easybuild/easyconfigs/s/STAR/STAR-2.7.10b-GCC-12.2.0.eb:33]:
```suggestion
    (['source/%(name)s', 'source/%(name)slong'], 'bin'),
```

File: easybuild/easyconfigs/s/STAR/STAR-2.7.10b-GCC-12.2.0.eb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    'CHANGES.md', 'doc', 'extras', 'LICENSE', 'README.md', 'RELEASEnotes.md',
+]
+
+sanity_check_paths = {
+    'files': ['bin/STAR', 'bin/STARlong'],
+    'dirs': [],
+}
+
```
Comment (jfgrimm) [inline easybuild/easyconfigs/s/STAR/STAR-2.7.10b-GCC-12.2.0.eb:41]:
```suggestion

sanity_check_commands = [
    "STAR --help",
    "STARlong --help",
]


```

**Review by jfgrimm — DISMISSED** [review]
for `STAR`, I've suggested some changes to pull it in line with the existing [STAR-2.7.10b-GCC-11.3.0.eb](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/s/STAR/STAR-2.7.10b-GCC-11.3.0.eb) easyconfig (which ensures the EasyBuild-built HTSlib is linked)

**Comment by eylenth** [comment]
> for `STAR`, I've suggested some changes to pull it in line with the existing [STAR-2.7.10b-GCC-11.3.0.eb](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/s/STAR/STAR-2.7.10b-GCC-11.3.0.eb) easyconfig (which ensures the EasyBuild-built HTSlib is linked)

done!

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by branfosj** [comment]
@boegelbot: please test @ generoso

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 3 out of 3 (2 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/355ff23280eac30c8b3e191b37eb09db for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Going in, thanks @eylenth!

**Pull request merged**